File: Packages/RallyCore/Sources/RallyCore/Theme/RallyColors.swift
import SwiftUI

/// Brand color constants for the Rally design system.
public enum RallyColors {
    /// Rally Orange — Primary brand color (#FF6B35)
    public static let orange = Color(red: 1.0, green: 0.42, blue: 0.21)

    /// Navy — Primary dark background (#131B2E)
    public static let navy = Color(red: 0.075, green: 0.106, blue: 0.18)

    /// Navy Mid — Secondary dark background (#1C2842)
    public static let navyMid = Color(red: 0.11, green: 0.157, blue: 0.259)

    /// Accent Blue — Secondary accent (#2D9CDB)
    public static let blue = Color(red: 0.176, green: 0.612, blue: 0.859)

    /// Off-White — Light background (#F5F7FA)
    public static let offWhite = Color(red: 0.961, green: 0.969, blue: 0.98)

    /// Medium Gray — Secondary text (#8B95A5)
    public static let gray = Color(red: 0.545, green: 0.584, blue: 0.647)

    /// Success Green
    public static let success = Color(red: 0.18, green: 0.8, blue: 0.44)

    /// Error Red
    public static let error = Color(red: 0.91, green: 0.27, blue: 0.33)

    /// Warning Yellow
    public static let warning = Color(red: 1.0, green: 0.79, blue: 0.19)
}

/// Color extension for theme-based access.
public extension Color {
    static let rally = RallyColorNamespace()
}

public struct RallyColorNamespace {
    public let orange = RallyColors.orange
    public let navy = RallyColors.navy
    public let navyMid = RallyColors.navyMid
    public let blue = RallyColors.blue
    public let offWhite = RallyColors.offWhite
    public let gray = RallyColors.gray
    public let success = RallyColors.success
    public let error = RallyColors.error
    public let warning = RallyColors.warning
}

File: Android/core/src/main/java/com/rally/app/core/theme/RallyColors.kt
package com.rally.app.core.theme

import androidx.compose.ui.graphics.Color

/**
 * Brand color constants for the Rally design system.
 *
 * These colors define the visual identity of the Rally platform
 * and serve as the foundation for both the default theme and
 * school-specific dynamic themes.
 */
object RallyColors {

    /** Rally Orange -- Primary brand color (#FF6B35) */
    val Orange = Color(0xFFFF6B35)

    /** Navy -- Primary dark background (#131B2E) */
    val Navy = Color(0xFF131B2E)

    /** Navy Mid -- Secondary dark background (#1C2842) */
    val NavyMid = Color(0xFF1C2842)

    /** Accent Blue -- Secondary accent (#2D9CDB) */
    val Blue = Color(0xFF2D9CDB)

    /** Off-White -- Light background (#F5F7FA) */
    val OffWhite = Color(0xFFF5F7FA)

    /** Medium Gray -- Secondary text (#8B95A5) */
    val Gray = Color(0xFF8B95A5)

    /** Success Green (#2ECC71) */
    val Success = Color(0xFF2ECC71)

    /** Error Red (#E84555) */
    val Error = Color(0xFFE84555)

    /** Warning Yellow (#FFC930) */
    val Warning = Color(0xFFFFC930)

    /** Pure White */
    val White = Color(0xFFFFFFFF)

    /** Pure Black */
    val Black = Color(0xFF000000)
}

/**
 * Parses a hex color string (e.g., "#FF6B35" or "FF6B35") into a Compose [Color].
 *
 * @param hex The hex color string, with or without leading '#'.
 * @return The parsed [Color], or null if the string is invalid.
 */
fun parseHexColor(hex: String): Color? {
    return try {
        val sanitized = hex.removePrefix("#")
        val colorLong = when (sanitized.length) {
            6 -> "FF$sanitized".toLong(16)
            8 -> sanitized.toLong(16)
            else -> return null
        }
        Color(colorLong)
    } catch (_: NumberFormatException) {
        null
    }
}
